projects/step-core/src/lib/modules/custom-registeries/services/base-registry.service.ts

```diff
@@ -0,0 +1,33 @@
+import { CustomRegistryType } from '../shared/custom-registry-type.enum';
+import { CustomRegistryService } from './custom-registry.service';
+import { Type } from '@angular/core';
+import { CustomRegistryItem } from '../shared/custom-registry-item';
+
+type ItemInfo = Pick<CustomRegistryItem, 'type' | 'label'>;
+
+const convert = (item?: CustomRegistryItem): ItemInfo | undefined => {
```

Comment:
a general question: what is the advantage of defining this piece of code outside of the Service? Why not define it as a private method?

Reply:
In current case `covert` is a very simple pure function that extract type and label, and it doesn't manipulate with service's data. So in my understanding, it doesn't belong to service.

In case if it would meet once in the service's code, it would be inline arrow function, but as it is appeared twice it was moved separately to prevent the code duplication.